src/usdb_syncer/notes_parser.py

```diff
@@ -438,20 +453,104 @@ def minimum_song_length(self) -> str:
 
         return f"{minutes:02.0f}:{seconds:02.0f}"
 
-    def set_first_timestamp_to_zero(self) -> None:
+    def fix_first_timestamp(self) -> None:
         """Shifts all notes such that the first note starts at beat zero and adjusts
         GAP accordingly
         """
         if (offset := self.notes.start()) == 0:
+            # round GAP to nearest 10 ms
+            self.headers.gap = round(self.headers.gap / 10) * 10
             return
         for line in self.notes.all_lines():
             line.shift(-offset)
-        self.headers.gap = self.headers.gap + beats_to_ms(offset, self.headers.bpm)
+        self.headers.gap = (
+            round((self.headers.gap + beats_to_ms(offset, self.headers.bpm)) / 10) * 10
+        )
+
+    def fix_low_bpm(self) -> None:
+        """(repeatedly) multiplies BPM value and all note timings by two
+        until the BPM is in a range (200, 400)"""
+        bpm_threshold = 200.0
+
+        if (original_bpm := self.headers.bpm) >= bpm_threshold:
+            return
+
+        while self.headers.bpm < bpm_threshold:
+            self.headers.bpm *= 2
+
+        timing_factor = int(self.headers.bpm / original_bpm)
+
+        # modify medley tags
+        if self.headers.medleystartbeat:
+            self.headers.medleystartbeat *= timing_factor
+
+        if self.headers.medleyendbeat:
+            self.headers.medleyendbeat *= timing_factor
+
+        # modify all note timings
+        for line in self.notes.all_lines():
+            line.multiply(timing_factor)
+
+    def fix_line_breaks(self) -> None:
+        for i, line in enumerate(self.notes.player_1):
+            if line.line_break:
+                # remove end
+                line.line_break.end = None
+                # set line break timing similar to USDX implementation
```

Comment:
Could you add a link to that part in USDX's source code?

Reply:
How to do that with the line limit length = 88?

https://github.com/UltraStar-Deluxe/USDX/blob/0974aadaa747a5ce7f1f094908e669209641b5d4/src/screens/UScreenEditSub.pas#L2976

Reply:
You can add ` # pylint: disable=line-too-long` to the end of the line.